test/include/kernels/GenericValueTest.h

```diff
@@ -0,0 +1,28 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "GenericKernel.h"
+
+template <bool is_ad>
+class GenericValueTestTempl : public GenericKernel<is_ad>
+{
+public:
+  static InputParameters validParams();
+
+  GenericValueTestTempl(const InputParameters & parameters);
+
+protected:
+  virtual GenericReal<is_ad> computeQpResidual();
+  virtual Real computeQpJacobian();
```

Comment:
I don't like this. I had suggested to put a 

```
virtual ADReal computeQpJacobian() {}
```

method in `GenericKernel<true>`, which would allow using `override` here:

```
protected:
  GenericReal<is_ad> computeQpResidual() override;
  GenericReal<is_ad> computeQpJacobian() override;
```

Reply:
I take that back. @vanwdani pointed out to me that the method already exists in `ADKernel` with this signature.

Reply:
The previous declaration of `virtual Real computeQpJacobian() {return 0;}` in `ADKernel` prevents this, but the override will be used.